src/components/v2/appDetails/sourceInfo/environmentStatus/AppStatusDetailModal.tsx

```diff
@@ -95,6 +129,21 @@ function AppStatusDetailModal({ close, appStreamData, showAppStatusMessage }: {
         }
     }
 
+    const onFilterClick = (selectedFilter: string): void => {
+        if (currentFilter !== selectedFilter) {
+            setCurrentFilter(selectedFilter)
+            if (selectedFilter === 'ALL') {
+                setFilteredNodes(flattenedNodes)
+            } else {
+                setFilteredNodes(
+                    flattenedNodes.filter(
+                        (nodeDetails) => nodeDetails.health?.status?.toLowerCase() === selectedFilter.toLowerCase(),
```

Comment:
Do we need an optional check here? If yes the please please add the same while sorting at line 54.

Reply:
Done